server/build.gradle

```diff
@@ -148,6 +147,9 @@ dependencies {
   api "org.apache.logging.log4j:log4j-core:${versions.log4j}", optional
   annotationProcessor "org.apache.logging.log4j:log4j-core:${versions.log4j}"
 
+  // gson
+  api 'com.google.code.gson:gson:2.10.1'
```

Comment:
To expand on this please look into the compatibility logs for what failed when they attempted to build, it might be as easy as a version number update on gson or it might be more complex.

![image](https://github.com/opensearch-project/OpenSearch/assets/2754967/828cde84-71ca-415c-9eae-6a711afbd6a4)

#### Links to sections of that doc:

- SQL https://github.com/opensearch-project/OpenSearch/actions/runs/7100520996/job/19326777156?pr=11441#step:4:14126
- Reporting https://github.com/opensearch-project/OpenSearch/actions/runs/7100520996/job/19326777156?pr=11441#step:4:12319

Reply:
I bumped the gson version in reporting and sql projects locally to 2.10.1 and they seem to be building fine with ./gradlew assemble. Will follow up with a PR for those projects. Not sure if there might be compatibility check failures (downstream build break) for them though.